lib/Remote/RemoteWebElement.php

```diff
@@ -371,6 +388,18 @@ public function setFileDetector(FileDetector $detector)
      */
     public function submit()
     {
+        if ($this->w3cCompliant) {
+            $this->executor->execute(DriverCommand::EXECUTE_SCRIPT, [
+                'script' => sprintf('return arguments[0]%s.submit();', 'form' === $this->getTagName() ? '' : '.form'),
+                'args' => [[JsonWireCompat::WEB_DRIVER_ELEMENT_IDENTIFIER => $this->id]],
+            ]);
+            // Wait for the new page to be ready
+            // TODO: wait for <html> to be ready, or use another more elegant way
+            usleep(10000);
```

Comment:
Why is this here? Isn't possible to `wait()`? This won't be reliable IMHO.

Reply:
Unfortunately I tried to use wait() but IIRC it's too early. I'll take a deeper look.